### ISSUE TITLE
Jumping Steeljaw Snapper off the coast of Tanaris

### ISSUE THREAD

**Issue by ghost** [azerothcore/azerothcore-wotlk#6166]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [x] 40-49
- [ ] 50-59

Original report: https://github.com/chromiecraft/chromiecraft/issues/750

##### CURRENT BEHAVIOUR:

I was fishing off the coast of Tanaris (near Steamwheedle Port, at 64.6 ; 24.6) and a Steeljaw Snapper started jumping. Seems like an issue with water/terrain?
Here is a video: https://youtu.be/Os8JN2G27Mg

##### EXPECTED BLIZZLIKE BEHAVIOUR:

The turtle should be swimming or walking, not jumping up and down.

##### STEPS TO REPRODUCE THE PROBLEM:

1. Go to 67.4 ; 24.6 in Tanaris, wait a moment till the Steeljaw Snapper jumps.

##### EXTRA NOTES:

N/A.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/3716ddf3e4e1ef138df334d01980fea019272001

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

Confirmed and reproduced on a local AC server (AC HASH/COMMIT: ea76481):

![WoWScrnShot_060121_044426](https://user-images.githubusercontent.com/82579073/120255385-7c392400-c294-11eb-99b5-303f4410721d.jpg)
![WoWScrnShot_060121_044432](https://user-images.githubusercontent.com/82579073/120255451-a5f24b00-c294-11eb-8a58-dbe881091f15.jpg)
![WoWScrnShot_060121_044518](https://user-images.githubusercontent.com/82579073/120255508-c9b59100-c294-11eb-94f5-0582ea9c549d.jpg)

In the photo it is not appreciated well, but I can confirm that it happens.

**Comment by claudiodfc** [comment]
@UltraNix Do you know if this is related to Vmaps? Could it be that the the Turtle size and the shallow water is trying to place it in ground but fails a check?

**Comment by T1ti** [comment]
I see similar stuff happen even without water, I think that's when the clien map deosn't match the virtual map, the creature being sent a coordinate from the server that deosn't match the physical map, it keeps visually jumping becasue the client physics deosn't match

**Comment by T1ti** [comment]
I saw yesterday a core hound in molten core do that on the thin bridge, specifically

**Comment by claudiodfc** [comment]
More vmaps issues then

**Comment by UltraNix** [comment]
No. Check if this creature have `UNIT_FLAG_SWIMMING` or/and `Inhabit_water` in its `creature_template` entry.

**Comment by claudiodfc** [comment]
Yes, they have both `UNIT_FLAG_SWIMMING ` and the correct `Inhabit_water` both water and ground.

**Comment by claudiodfc** [comment]
This should be the correct behaviour when swapping form one inhabit type to the other:
https://www.youtube.com/watch?v=VlapUuuyVy4&ab_channel=AYeeAYee
